File: app/src/main/java/rod/bailey/trafficatnsw/json/hazard/HazardCollection.java
package rod.bailey.trafficatnsw.json.hazard;

import com.google.gson.Gson;

import java.util.Date;
import java.util.List;

/**
 * Represents the entire JSON file containing multiple hazards.
 */
public class HazardCollection {
    public List<Hazard> features;
    public long lastPubished;

    public static HazardCollection parseJson(String jsonContents) {
        Gson gson = new Gson();
        return gson.fromJson(jsonContents, HazardCollection.class);
    }
}

File: app/src/androidTest/java/rod/bailey/trafficatnsw/json/hazard/HazardCollection.kt
package rod.bailey.trafficatnsw.json.hazard

import com.google.gson.Gson

/**
 * Represents the entire JSON file containing multiple hazards.
 */
data class HazardCollection(var features: List<Hazard>? = null, var lastPublished: Long = 0L) {

	fun findHazardById(hazardId: Int): Hazard? {
		if (features != null) {
			for (hazard in features!!) {
				if (hazard.id == hazardId) {
					return hazard
				}
			}
		}
		return null
	}

	companion object {

		fun parseJson(jsonContents: String): HazardCollection {
			val gson = Gson()
			return gson.fromJson(jsonContents, HazardCollection::class.java)
		}
	}
}
